include/envoy/stats/stats.h

```diff
@@ -94,5 +94,18 @@ class Gauge : public virtual Metric {
 
 typedef std::shared_ptr<Gauge> GaugeSharedPtr;
 
+/**
+ * A string, possibly non-ASCII.
+ */
+class TextReadout : public virtual Metric {
```

Comment:
How about TextGauge?

Reply:
Well, this new class is meant to remove the need to abuse gauges, so in some sense it's explicitly not a gauge. The name "gauge" suggests the image of a circle with numbers and a needle bouncing around, like what comes up in a Google image search for "gauge".

Reply:
:-) . Agree. It sounds more like a numeric thing.

Reply:
Do we want this to indicate that it s a stat? If yes, may be 

`TextMetric`/`TextStat`/`TextValuedStat` may convey it more obviously?

Another thing, I was looking statsd and prometheus documentation and they do not seem to support text type of stats - so if we  `migrate the 64-bit hash gauges` to this do we have to worry about the existing users?